conference/src/test/java/org/spine/samples/lobby/conference/ConferenceServiceShould.java

```diff
@@ -212,5 +219,141 @@ public Void apply(Void input) {
+            @Override
+            public Void apply(final PublishConferenceResponse response) {
+                final FindConferenceByIDRequest request = FindConferenceByIDRequest.newBuilder()
+                                                                                   .setId(response.getId())
+                                                                                   .build();
+                conferenceService.findConferenceByID(request, TestStreamObserver.<Conference>newBuilder()
+                                                                                .setNextFunction(checkIsPublished)
+                                                                                .build());
+
+
+                return NO_RESULT;
+            }
+        };
+        final PublishConferenceRequest request = PublishConferenceRequest.newBuilder()
+                                                                         .setId(conference.getId())
+                                                                         .build();
+
+        conferenceService.publish(request, TestStreamObserver.<PublishConferenceResponse>newBuilder()
+                                                             .setNextFunction(publishConferenceHandler)
+                                                             .build());
+
+        final EventStore eventStore = boundedContext.getEventBus()
+                                                    .getEventStore();
+
+        final Function<Event, Void> checkConferencePublished = new Function<Event, Void>() {
+            @SuppressWarnings("ReturnOfNull")
+            @Override
+            public Void apply(Event event) {
+                accumulatedMessages.add(Messages.fromAny(event.getMessage())
+                                                .getClass());
+                return NO_RESULT;
+            }
+        };
+
+
+        final Function<Void, Void> onCompleteUpdate = new Function<Void, Void>() {
+            @Override
+            public Void apply(Void input) {
+                assertTrue(accumulatedMessages.contains(ConferencePublished.class));
+                return NO_RESULT;
+            }
+        };
+
+        final TestStreamObserver conferencePublishedObserver = TestStreamObserver.<Event>newBuilder()
+                                                                                 .setNextFunction(checkConferencePublished)
+                                                                                 .setOnCompleteFunction(onCompleteUpdate)
+                                                                                 .build();
+
+        eventStore.read(EventStreamQuery.newBuilder()
+                                        .build(), conferencePublishedObserver);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void unpublish_conference_and_generate_ConferenceUnpublished_event() {
```

Comment:
I'm pretty sure you know this already, and just don't pay enough attention to the quality of test code, as a lot of people do, and I did as well. But professionals definitely say that the test code quality is as important as the production code quality.